### PR TITLE
Added possibility to add listeners

### PR DESCRIPTION
**Pull request by SvetlovA**
This adds possibility to use custom listeners and existing listeners like DataLoaderDocumentListener to use DataLoader

### CODE REVIEW

**Comment by SvetlovA** [comment]
Can you see my pull request or write me how can i solve N+1 problem without this solution, if it possible?

File: src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs [hunk 1]
```diff
@@ -256,9 +257,15 @@ public string SerializeResult(ExecutionResult result)
                 UserContext = UserContextWrapper.Create(userContext, dependencyInjector ?? new WrappedDependencyInjector(_constructor.TypeResolutionDelegate)),
                 ValidationRules = rules != null && rules.Any() ? rules : null,
                 ComplexityConfiguration = complexityConfiguration,
-                CancellationToken = cancellationToken,
+                CancellationToken = cancellationToken
             };
 
+            if (listeners != null && listeners.Any())
```
Comment (Igorbek) [inline src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs:263]: ` && listeners.Any()` is redundant and causes double enumeration. If `listeners` is empty nothing would be added.

**Review by Igorbek — CHANGES_REQUESTED** [review]

**Review by tlil — APPROVED** [review]

**Comment by tlil** [comment]
Sorry for the late review - thanks!

**Comment by huysentruitw** [comment]
I'm creating a sample application with EF Core + DataLoader support and will send a PR.

**Comment by tlil** [comment]
Nice, sounds good @huysentruitw

**Pull request merged**